webapp/src/clj/lipas/backend/legacy/api.clj

```diff
@@ -39,10 +39,30 @@
       (localize lang)
       utils/->camel-case-keywords))
 
-(defn sports-place-types [lang]
+(defn sports-place-types [locale]
   (->>  (vals types/all)
-        (map #(->legacy-api % lang))))
+        (map #(->legacy-api % locale))))
 
-(defn sports-place-by-type-code [lang type-code]
+(defn sports-place-by-type-code [locale type-code]
   (-> (types/all type-code)
-      (->legacy-api-with-properties lang)))
+      (->legacy-api-with-properties locale)))
+
+(defn collect-sport-place-types [sub-category-type-code] 
+  (->> (vals types/all)
+       (filter #(= (% :sub-category) sub-category-type-code))
```

Comment:
Minor thing, but you can change `filter` to just a map lookup. We already have types indexed by sub-category (and also by main-category) here https://github.com/lipas-liikuntapaikat/lipas/blob/remove-legacy-api-v2/webapp/src/cljc/lipas/data/types.cljc#L23-L24